### PR TITLE
Add four Utah credits

### LINKED ISSUE #957: Utah EITC
[15% nonrefundable](https://le.utah.gov/xcode/Title59/Chapter10/59-10-S1044.html?v=C59-10-S1044_2022050420220504)

### PR DESCRIPTION
**Pull request by nikhilwoodruff**
Fixes #957

### CODE REVIEW

File: policyengine_us/variables/gov/states/ut/tax/income/credits/retirement_credit/ut_claims_retirement_credit.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    value_type = bool
+    entity = TaxUnit
+    label = "claims the Utah retirement credit"
+    unit = USD
+    documentation = "Utah residents can claim only one of the retirement credit or the social security benefits credit. We assume they claim the one with higher value."
+    definition_period = YEAR
+    defined_for = StateCode.UT
+
+    def formula(tax_unit, period, parameters):
+        max_retiement_credit = tax_unit("ut_retirement_credit_max", period)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ut/tax/income/credits/retirement_credit/ut_claims_retirement_credit.py:14]:
```suggestion
        max_retirement_credit = tax_unit("ut_retirement_credit_max", period)
```

File: policyengine_us/parameters/gov/states/ut/tax/income/credits/taxpayer/refundable.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ut/tax/income/credits/taxpayer/refundable.yaml:1]: This is a different pattern than what we've done in other states with a list of refundable/non-refundable credits

**Review by MaxGhenis — COMMENTED** [review]
could you add unit tests?

File: policyengine_us/parameters/gov/states/ut/tax/income/credits/taxpayer/refundable.yaml
Comment (nikhilwoodruff) [reply policyengine_us/parameters/gov/states/ut/tax/income/credits/taxpayer/refundable.yaml:1]: Ah, yes... do we need to change it? Would have no impact and I think this is closer to how the tax form works.

**Comment by nikhilwoodruff** [comment]
@MaxGhenis added unit tests for the variables.

**Pull request merged**